File: order-producer/src/main/java/com/sahakav/coffee_shop/resource/OrderCommandResource.java
package com.sahakav.coffee_shop.resource;

import com.sahakav.coffee_shop.model.CoffeeType;
import com.sahakav.coffee_shop.service.OrderCommandService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.UUID;

@RestController
public class OrderCommandResource {
    private final OrderCommandService orderCommandService;


    public OrderCommandResource(OrderCommandService orderCommandService) {
        this.orderCommandService = orderCommandService;
    }

    @PostMapping(name = "/order", consumes = "application/json", produces = "application/json")
    public ResponseEntity<UUID> placeOrder(){
        UUID uuid = orderCommandService.placeOrder(CoffeeType.valueOf("LATTE")).get();
        return ResponseEntity.accepted().body(uuid);
    }
}

File: order-producer/src/main/scala/com/sahakav/coffee_shop/resource/OrderCommandResource.scala
package com.sahakav.coffee_shop.resource

import java.util.UUID

import com.sahakav.coffee_shop.model.{CoffeeOrderState, CreateCoffeeOrderRequestModel}
import com.sahakav.coffee_shop.service.OrderCommandService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.{GetMapping, PathVariable, PostMapping, RequestBody, RequestMapping, RequestMethod, RestController}

import scala.util.{Failure, Success}

@RestController
@RequestMapping()
class OrderCommandResource (@Autowired val orderCommandService: OrderCommandService){


  @RequestMapping(path = Array("/order"), consumes = Array("application/json"), produces = Array("application/json"), method = Array(RequestMethod.POST))
  def placeOrder(@RequestBody createCoffeeOrderDto: CreateCoffeeOrderRequestModel): ResponseEntity[UUID] = {
     orderCommandService.placeOrder(createCoffeeOrderDto.coffeeType) match {
       case Success(value) => ResponseEntity.accepted().body(value)
       case Failure(exception) => throw new RuntimeException(exception)
     }
  }

  @RequestMapping(path = Array("/order/{id}"),  produces = Array("application/json"), method = Array(RequestMethod.GET))
  def getOrderState(@PathVariable(name = "id") id:UUID): CoffeeOrderState ={
    orderCommandService.getOrderState(id)
  }
}
